src/nvim/api/vim.c

```diff
@@ -990,6 +990,42 @@ void nvim_set_option(uint64_t channel_id, String name, Object value, Error *err)
   set_option_to(channel_id, NULL, SREQ_GLOBAL, name, value, err);
 }
 
+/// Echo a message.
+///
+/// @param str Message
+/// @param opts Optional parameters. Keys:
+///          - hl_group: name of the highlight group used to highlight.
+///          - history: if true, add to |message-history|.
+void nvim_echo(String str, Dictionary opts, Error *err)
+  FUNC_API_SINCE(7)
+{
+  int hl_id = 0;
+  bool history = false;
```

Comment:
we probably want `history` to be the default (or make it mandatory arg even?)

Reply:
> we probably want history to be the default

right.
come to think of it, I usually use `echomsg` in my plugin's message.